### PR TITLE
[16.0][FIX] sale_order_carrier_auto_assign: fix all service

### PR DESCRIPTION
**Pull request by jbaudoux**
Do not add carrier when SO contains only service lines

Also increased test coverage

Removed Alpha status

@sebalix @simahawk @santostelmo

### CODE REVIEW

**Review by santostelmo — APPROVED** [review]

File: sale_order_carrier_auto_assign/__manifest__.py [hunk 1]
```diff
@@ -4,10 +4,10 @@
     "name": "Sale Order Carrier Auto Assign",
     "summary": "Auto assign delivery carrier on sale order confirmation",
     "version": "16.0.1.1.0",
-    "development_status": "Alpha",
```
Comment (sebalix) [inline sale_order_carrier_auto_assign/__manifest__.py:7]: If it's not `Alpha`, is it `Beta`/`Mature`/`Production/Stable`?
I think we should not remove it, I do not know the default value...?

Comment (jbaudoux) [reply sale_order_carrier_auto_assign/__manifest__.py:7]: Default is Beta

**Comment by jbaudoux** [comment]
@sebalix Can we merge this?

**Comment by jbaudoux** [comment]
ping @sebalix

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**